### PR TITLE
#11126: Resnet Demo with new conv API

### PR DESCRIPTION
**Pull request by sankarmanoj-tt**
### Ticket
#11126

### Problem description
Implement Resnet Demo using the new Conv2D API

### Checklist
- [x] Post commit CI [passes](https://github.com/tenstorrent/tt-metal/actions/runs/10529442898)

### CODE REVIEW

File: models/demos/ttnn_resnet/demo/demo.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
```
Comment (tt-aho) [inline models/demos/ttnn_resnet/demo/demo.py:1]: Use 2024?

**Review by tt-aho — COMMENTED** [review]
Can you update the readme with the instructions to run the demo similar to what's in `models/demos/resnet/README.md`?

File: models/demos/ttnn_resnet/tt/ttnn_functional_resnet50_new_conv_api.py [hunk 2]
```diff
@@ -575,7 +575,9 @@ def __init__(
             compute_kernel_config=compute_kernel_config,
         )  # num_classes = 1000
 
-        self.transpose_shards = True
+        self.transpose_shards = False
+        act_block_h_override = 1568
```
Comment (mywoodstock) [inline models/demos/ttnn_resnet/tt/ttnn_functional_resnet50_new_conv_api.py:579]: these changes needed?

**Review by mywoodstock — COMMENTED** [review]
Please also update the README for this demo

File: models/demos/ttnn_resnet/tt/ttnn_functional_resnet50_new_conv_api.py (hunk 2 above)
Comment (sankarmanoj-tt) [reply models/demos/ttnn_resnet/tt/ttnn_functional_resnet50_new_conv_api.py:579]: Yes. I was getting an error that `act_block_h_override` was not defined. I think `is_wormhole_b0()` was False.

File: models/demos/ttnn_resnet/demo/demo.py (hunk 1 above)
Comment (sankarmanoj-tt) [reply models/demos/ttnn_resnet/demo/demo.py]: Done.

**Comment by sankarmanoj-tt** [comment]
@tt-aho README has been updated.

File: isp.py
Comment (tt-aho) [inline isp.py:1]: Is this a mistakenly added file?

**Review by mywoodstock — APPROVED** [review]

File: isp.py
Comment (tt-aho) [inline isp.py:1]: Delete this file?

File: models/demos/ttnn_resnet/README.md [hunk 3]
```diff
@@ -5,14 +5,25 @@
...
+; where 16 is the batch size, and `models/demos/resnet/demo/images/` is where the images are located. Our model supports batch size of 2 and 1 as well, however the demo focuses on batch size 16 which has the highest throughput among the three options. This demo includes preprocessing, postprocessing and inference time for batch size 16. The demo will run the images through the inference thrice. First, discover the optimal shard scheme. Second to capture the compile time, and cache all the ops. Thrid, to capture the best inference time on TT hardware.
+
++ Our second demo is designed to run ImageNet dataset, run this with
+`pytest --disable-warnings models/demos/ttnn_resnet/demo/demo.py::test_demo_imagenet[16-100-device_params0]`; again 16 refer to batch size here and 100 is number of iterations(batches), hence the model will process 100 batch of size 16, total of 1600 images.
+
+Note that the first time the model is run, ImageNet images must be downloaded from huggingface and stored in  'models/demos/resnet/demo/images/'; therefore you need to login to huggingface using your token: 'huggingface-cli login' or by setting the token with the command `export HF_TOKEN=<token>`
```
Comment (tt-aho) [inline models/demos/ttnn_resnet/README.md:23]:
```suggestion
Note that the first time the model is run, ImageNet images must be downloaded from huggingface and stored in  `models/demos/resnet/demo/images/`; therefore you need to login to huggingface using your token: `huggingface-cli login` or by setting the token with the command `export HF_TOKEN=<token>`
```

File: models/demos/ttnn_resnet/README.md [hunk 4]
```diff
@@ -5,14 +5,25 @@
...
 
 + The entry point to the Metal ResNet model is `ResNet` in `ttnn_functional_resnet50_new_conv_api.py`.
 + The model picks up certain configs and weights from TorchVision pretrained model. We have used `torchvision.models.ResNet50_Weights.IMAGENET1K_V1` version from TorchVision as our reference.
 + Our ImageProcessor on the other hand is based on `microsoft/resnet-50` from huggingface.
 
+## Demo
+
++ To run the demo use:
+`pytest --disable-warnings models/demos/ttnn_resnet/demo/demo.py::test_demo_sample[16-models/demos/resnet/demo/images/-device_params0]`
+; where 16 is the batch size, and `models/demos/resnet/demo/images/` is where the images are located. Our model supports batch size of 2 and 1 as well, however the demo focuses on batch size 16 which has the highest throughput among the three options. This demo includes preprocessing, postprocessing and inference time for batch size 16. The demo will run the images through the inference thrice. First, discover the optimal shard scheme. Second to capture the compile time, and cache all the ops. Thrid, to capture the best inference time on TT hardware.
```
Comment (tt-aho) [inline models/demos/ttnn_resnet/README.md:18]:
```suggestion
; where 16 is the batch size, and `models/demos/resnet/demo/images/` is where the images are located. Our model supports batch size of 2 and 1 as well, however the demo focuses on batch size 16 which has the highest throughput among the three options. This demo includes preprocessing, postprocessing and inference time for batch size 16. The demo will run the images through the inference thrice. First, discover the optimal shard scheme. Second to capture the compile time, and cache all the ops. Third, to capture the best inference time on TT hardware.
```

**Review by tt-aho — CHANGES_REQUESTED** [review]

**Review by tt-aho — APPROVED** [review]

**Pull request merged**